docs-developer/references/i18n.rst

```diff
@@ -81,5 +81,4 @@ These messages will then be discovered for any registered plugins and loaded int
 Adding a new language
 ---------------------
 
-In order to add a new supported language to Kolibri, the appropriate language tuple must be added to the ``LANGUAGES`` variable in ``kolibri/deployment/default/settings/base.py``. In addition, the appropriate Intl polyfill file must be added to ``kolibri/core/assets/src/utils/import-intl-locale.js``.
-
+In order to add a new supported language to Kolibri, the appropriate language information object must be added to the array in ``kolibri/locale/supported_languages.json``.
```

Comment:
Yes, unfortunately, the only price you pay is :magic:.

Reply:
This section is about frontend translation only, right?

Reply:
No, previously we were specifying information about supported languages in two places, the Django settings and the frontend intl polyfill locale data.

With this PR we would now be encoding the supported languages in three places, with the need to add vue intl locale data. 

Rather than maintain three separate representations of which languages are supported (which had already gotta out of sync at one point), I have abstracted everything into the supported locales json. This is then used to generate the languages setting in django, and add the appropriate locale data to the frontend.

Reply:
It's not possible to use the Django settings to populate the frontend? Backend supporting the frontend just seems somehow more intuitive here...

On another note: There is a strong chance that a tailored deployment will want to specify languages available to the user. Would this be obstructing or is this obstructed in any way? I could see that a deployment would set its own Django `settings.LANGUAGES` and then that would configure everything.. but it seems that the frontend wouldn't pick it up then? Or is this only about generating translation locales? Or both?

Reply:
The frontend, at run time, is entirely driven by the django languages setting.

The supported locales are what we need to build our frontend code to support. If a user chooses only to enable some subset of these, that is their perogative and Kolibri will work perfectly.

Reply:
Awesome, thanks 👍